includes/events/class-omise-event-charge-complete.php

```diff
@@ -44,68 +44,26 @@ class Omise_Event_Charge_Complete {
 	 * @return void
 	 */
 	public function handle( $data ) {
-		if ( 'charge' !== $data->object || ! isset( $data->metadata->order_id ) ) {
+		if ( 'charge' !== $data['object'] || ! isset( $data['metadata']['order_id'] ) ) {
 			return;
 		}
 
-		if ( ! $order = wc_get_order( $data->metadata->order_id ) ) {
+		if ( ! $order = wc_get_order( $data['metadata']['order_id'] ) ) {
 			return;
 		}
 
 		// Making sure that an event's charge id is identical with an order transaction id.
-		if ( $order->get_transaction_id() !== $data->id ) {
+		if ( $order->get_transaction_id() !== $data['id'] ) {
 			return;
 		}
 
-		$order->add_order_note(
-			__(
-				'Omise: an event charge.complete has been caught (webhook).',
-				'omise'
-			)
-		);
-
-		switch ($data->status) {
```

Comment:
Note: moving the entire order-status-manipulation logic to a dedicated class, `Omise_Payment_Result`. As the Callback and the Webhook event are sharing the same code.